addons/l10n_be_intrastat/wizard/xml_decl.py

```diff
@@ -286,3 +291,20 @@ def _set_Dim(self, item, prop, value):
         dim = ET.SubElement(item, 'Dim')
         dim.set('prop', prop)
         dim.text = value
+
+    def _is_situation_triangular(self, company, po_line=False, so_line=False):
+        # Ignoring what is purchased and sold by us with a dropshipping route
+        # outside of our country
+        # Freight did not set foot on our territory and it shall not be reported
+        dropship_pick_type = self.env.ref('stock_dropshipping.picking_type_dropship', raise_if_not_found=False)
+        if not dropship_pick_type:
+            return False
+        stock_move_domain = [('picking_type_id', '=', dropship_pick_type.id)]
+
+        if po_line:
+            stock_move_domain.append(('purchase_line_id', '=', po_line.id))
+        if so_line:
+            stock_move_domain.append(('procurement_id.sale_line_id', '=', so_line.id))
+
+        stock_move = self.env['stock.move'].search(stock_move_domain, limit=1)
+        return stock_move and stock_move.partner_id.country_id.code != company.country_id.code and stock_move.picking_partner_id.country_id.code != company.country_id.code
```

Comment:
mmh I'm not sure that's the best way to check if the situation is triangular: I would have tested if source location type is supplier and destination location type is customer

That being said, I don't know what you need to do with sales/purchases made from a belgian supplier to a belgian customer in dropshipping. Should such operations be reported? Are regular sales/purchase within Belgium included currently? I've no idea...

Reply:
@qdp-odoo 
Checking location source = supplier and dest = customer only tests if we are in dropshipping
I suppose I can do that in my stock.move domain

For the rest of the use cases you mention, the 2017 manual doesn't say much (and I think we are free to implement belgium to belgium flows)
https://www.nbb.be/doc/dq/f_pdf_ex/basis2017fr.pdf 
pages 15-16

Reply:
@kebeclibre 
> Checking location source = supplier and dest = customer only tests if we are in dropshipping

I though that was exactly what you wanted to test. The only thing with that approach is that it will exclude belgium-to-belgium dropships whereas maybe we're including belgium-to-belgium regular sales/purchases, which would be inconsistent. @nseinlet might have a clue...

Reply:
@qdp-odoo 
I want to test if we are in dropshipping and outside of our borders

this is what is done with (this returns false if at least one of the partner is in belgium)
>> stock_move.partner_id.country_id.code != company.country_id.code and stock_move.picking_partner_id.country_id.code != company.country_id.code

Reply:
Intrastat means crossing borders inside EU. So, one of the involved partner should be in another EU country.
Triangular means from a country, to another, even without going through the third partner country.
https://www.nbb.be/doc/dq/e_pdf_ex/basis2015en.pdf page 6-2